smithy-aws-protocol-tests/model/restXml/http-query.smithy

```diff
@@ -351,3 +363,119 @@ structure QueryIdempotencyTokenAutoFillInput {
     @idempotencyToken
     token: String,
 }
+
+// Clients must make named query members take precedence over unnamed members
+// and servers must use all query params in the unnamed map.
+@http(uri: "/Precedence", method: "POST")
+operation QueryPrecedence {
+    input: QueryPrecedenceInput
+}
+
+apply QueryPrecedence @httpRequestTests([
+    {
+        id: "RestXmlQueryPrecedence",
+        documentation: "Prefer named query parameters when serializing",
+        protocol: restXml,
+        method: "POST",
+        uri: "/Precedence",
+        body: "",
+        queryParams: [
+            "bar=named",
+            "qux=alsoFromMap"
+        ],
+        params: {
+            foo: "named",
+            baz: {
+                foo: "fromMap",
```

Comment:
Does this key need to be `bar`? It is in the next test.

Reply:
Fixed in f4a20c4.